pkg/api/v20170701/types.go

```diff
@@ -0,0 +1,294 @@
+// where:
+//    <SUB_ID> is the subscription ID of the keyvault
+//    <RG_NAME> is the resource group of the keyvault
+//    <KV_NAME> is the name of the keyvault
+//    <NAME> is the name of the secret.
+//    <VERSION> (optional) is the version of the secret (default: the latest version)
+type ServicePrincipalProfile struct {
+	ClientID string `json:"clientId,omitempty"`
+	Secret   string `json:"secret,omitempty"`
+}
+
+// CertificateProfile represents the definition of the master cluster
+// The JSON parameters could be either a plain text, or referenced to a secret in a keyvault.
+// In the latter case, the format of the parameter's value should be
+// "/subscriptions/<SUB_ID>/resourceGroups/<RG_NAME>/providers/Microsoft.KeyVault/vaults/<KV_NAME>/secrets/<NAME>[/<VERSION>]"
+// where:
+//    <SUB_ID> is the subscription ID of the keyvault
+//    <RG_NAME> is the resource group of the keyvault
+//    <KV_NAME> is the name of the keyvault
+//    <NAME> is the name of the secret
+//    <VERSION> (optional) is the version of the secret (default: the latest version)
+type CertificateProfile struct {
+	// CaCertificate is the certificate authority certificate.
+	CaCertificate string `json:"caCertificate,omitempty"`
+	// ApiServerCertificate is the rest api server certificate, and signed by the CA
+	APIServerCertificate string `json:"apiServerCertificate,omitempty"`
+	// ApiServerPrivateKey is the rest api server private key, and signed by the CA
+	APIServerPrivateKey string `json:"apiServerPrivateKey,omitempty"`
+	// ClientCertificate is the certificate used by the client kubelet services and signed by the CA
+	ClientCertificate string `json:"clientCertificate,omitempty"`
+	// ClientPrivateKey is the private key used by the client kubelet services and signed by the CA
+	ClientPrivateKey string `json:"clientPrivateKey,omitempty"`
+	// KubeConfigCertificate is the client certificate used for kubectl cli and signed by the CA
+	KubeConfigCertificate string `json:"kubeConfigCertificate,omitempty"`
+	// KubeConfigPrivateKey is the client private key used for kubectl cli and signed by the CA
+	KubeConfigPrivateKey string `json:"kubeConfigPrivateKey,omitempty"`
+	// caPrivateKey is an internal field only set if generation required
+	caPrivateKey string
+}
+
+// LinuxProfile represents the Linux configuration passed to the cluster
+type LinuxProfile struct {
+	AdminUsername string `json:"adminUsername"`
+
+	SSH struct {
+		PublicKeys []struct {
+			KeyData string `json:"keyData"`
+		} `json:"publicKeys"`
+	} `json:"ssh"`
+	Secrets []KeyVaultSecrets `json:"secrets,omitempty"`
+}
+
+// WindowsProfile represents the Windows configuration passed to the cluster
+type WindowsProfile struct {
+	AdminUsername string            `json:"adminUsername,omitempty"`
+	AdminPassword string            `json:"adminPassword,omitempty"`
+	Secrets       []KeyVaultSecrets `json:"secrets,omitempty"`
```

Comment:
Is WindowsProfile exposing KeyVaultSecrets in the public API model? If not then it should not be added here.